ansible/modules/hashivault/hashivault_approle_role.py

```diff
@@ -0,0 +1,184 @@
+        default: token
+    token:
+        description:
+            - token for vault
+        default: to environment variable VAULT_TOKEN
+    username:
+        description:
+            - username to login to vault.
+        default: to environment variable VAULT_USER
+    password:
+        description:
+            - password to login to vault.
+        default: to environment variable VAULT_PASSWORD
+    state:
+        description:
+            - present or absent
+        default: present
+    name:
+        description:
+            - role name.
+    mount_point:
+        description:
+            - mount point for role
+        default: approle
+    bind_secret_id:
+        description:
+            - Require secret_id to be presented when logging in using this AppRole.
+    bound_cidr_list:
+        description:
+            - Comma-separated string or list of CIDR blocks.
+    policies:
+        description:
+            - policies for the role.
+    secret_id_num_uses:
+        description:
+            - Number of times any particular SecretID can be used.
+    secret_id_ttl:
+        description:
+            - Duration after which any SecretID expires.
+    token_num_uses:
+        description:
+            - Number of times issued tokens can be used. A value of 0 means unlimited uses.
+    token_ttl:
+        description:
+            - Duration to set as the TTL for issued tokens and at renewal time.
+    token_max_ttl:
+        description:
+            - Duration after which the issued token can no longer be renewed.
+    period:
+        description:
+            - Duration of the token generated.
+    enable_local_secret_ids:
+        description:
+            - If set, the secret IDs generated using this role will be cluster local.
+'''
+EXAMPLES = '''
+---
+- hosts: localhost
+  tasks:
+    - hashivault_approle_role:
+        name: ashley
+    - hashivault_approle_role:
+        name: ashley
+        state: absent
+'''
+
+
+def main():
+    argspec = hashivault_argspec()
+    argspec['state'] = dict(required=False, choices=['present', 'absent'], default='present')
+    argspec['name'] = dict(required=True, type='str')
+    argspec['mount_point'] = dict(required=False, type='str', default='approle')
+    argspec['bind_secret_id'] = dict(required=False, type='bool', no_log=True)
+    argspec['bound_cidr_list'] = dict(required=False, type='list')
+    argspec['policies'] = dict(required=False, type='list', default=[])
+    argspec['secret_id_num_uses'] = dict(required=False, type='str')
+    argspec['secret_id_ttl'] = dict(required=False, type='str')
+    argspec['token_num_uses'] = dict(required=False, type='int')
+    argspec['token_ttl'] = dict(required=False, type='str')
+    argspec['token_max_ttl'] = dict(required=False, type='str')
+    argspec['period'] = dict(required=False, type='str')
+    argspec['enable_local_secret_ids'] = dict(required=False, type='bool')
+    module = hashivault_init(argspec)
```

Comment:
if you support check mode (see below) you can pass it to the `_init()`. ex: `hashivault_init(argspec, supports_check_mode)`